src/ifcgeom/infra_sweep_helper.cpp

```diff
@@ -14,6 +14,27 @@ namespace {
 	}
 }
 
+namespace {
+template <typename T, typename Cmp = std::less<T>>
```

Comment:
Does Cmp = std::less<T> assume tags have increasing value start to end along the profile? Examples have been A,B,C  and A,B,D,C. What if C,B,A and  C,D,B,A? or something more random Z, A, Q and Z, A, W, Q?